doc/melange.rst

```diff
@@ -4,49 +4,188 @@
 JavaScript Compilation With Melange
 ***********************************
 
+Introduction
+------------
+
 `Melange <https://github.com/melange-re/melange>`_ is a compiler from OCaml to
 JavaScript. Unlike js_of_ocaml, Melange works by translating OCaml compiler
 internal lambda representation to JS files. This allows to produce a single
 JavaScript file from each OCaml module. Melange can be installed with
-`opam, Esy, or Nix <https://github.com/melange-re/melange#installation>`_ package
-managers.
+`opam, Esy, or Nix <https://github.com/melange-re/melange#installation>`_
+package managers.
+
+Dune can build projects using Melange, and it allows to generate JavaScript
+files by defining a :ref:`melange-emit` stanza. Dune libraries can be used
+with Melange, by leveraging the ``(modes melange)`` field in the :ref:`library`
+stanza.
+
+To enable Melange support in a Dune project, specify the Melange language
+version in the :ref:`dune-project` file. Adding
```

Comment:
should we mention that this is only available if you have `(lang dune 3.6)` in your `dune-project`?